### PR TITLE
Modular logserver outputs

### PR DESCRIPTION
**Pull request by fortizc**
Creates a new system to allow modular outputs in logserver. The new system uses a base struct to define an output and internally logserver use a list to store the needed outputs (following the configuration).

The current PR includes:
* define a base structure for new outputs (logserver_out)
* rewrite all old outputs using the new system
* creates a small "utils" lib for common functions

### CODE REVIEW

**Comment by asac** [comment]
remember to add copyright everywhere to good content :)

**Comment by fortizc** [comment]
> remember to add copyright everywhere to good content :)

Right! I forgot

File: logserver/logserver_utils.c [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+			      js.plat, pv_log_level_name(js.lvl), js.src,
+			      (int)strlen(js.data.buf),
+			      js.data.buf) +
+		     1; // 0 byte
+
+	char *json = calloc(1, len);
+	if (!json)
+		return NULL;
+
+	snprintf(json, len, LOGSERVER_JSON_FORMAT, js.tsec, js.tnano, js.plat,
```
Comment (anibalportero) [inline logserver/logserver_utils.c:126]: I think we should use the pv_json_ser functions here. We can leave it for another PR if this is just moving things around

File: logserver/logserver_utils.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ */
+
+#ifndef LOGSERVER_UTILS_H
+#define LOGSERVER_UTILS_H
+
+#include "logserver_out.h"
+
+#include <stdbool.h>
+
+int logserver_open_logfile(const char *path);
```
Comment (anibalportero) [inline logserver/logserver_utils.h:30]: we should stick to the logserver_utils_ prefix as you have done in the rest of functions in .h files

**Review by anibalportero — COMMENTED** [review]
As this change is old, you should  go through all the changes suffered by logserver.h and logserver.c since you opened it. I think not all of them are here. For example, compare pv_logserver_stop in master with yours and you'll see what I mean
